lib/manageiq/api/common/paginated_response.rb

```diff
@@ -7,8 +7,8 @@ class PaginatedResponse
         def initialize(base_query:, request:, limit: nil, offset: nil)
           @base_query = base_query
           @request    = request
-          @limit      = (limit || 100).to_i
-          @offset     = (offset || 0).to_i
+          @limit      = (limit || 100).to_i.clamp(1, 1000)
+          @offset     = (offset || 0).to_i.clamp(0, Float::INFINITY)
```

Comment:
you can leave off the second number entirely I think.

Reply:
Nope, https://github.com/ManageIQ/more_core_extensions/blob/master/lib/more_core_extensions/core_ext/numeric/clamp.rb#L9

Reply:
Oh weird...

we should probably change clamp to follow the semantics for the new Ruby clamp method anyway, but even so, for now then, this is fine.